doc/supervised.rst

```diff
@@ -62,87 +58,71 @@ NCA(init='auto', max_iter=100, n_components=None,
-  do that, use the `get_metric` method.
-
->>> metric_fun = nca.get_metric()
->>> metric_fun([3.5, 3.6], [5.6, 2.4])
-0.4962707194621285
-
-- Alternatively, you can use `pair_score` to return the **score** between
-  pairs of points (the larger the score, the more similar the pair).
-  For Mahalanobis learners, it is equal to the opposite of the distance.
+We can now use the learned metric to **score** new pairs of points with ``pair_score``
+(the larger the score, the more similar the pair). For Mahalanobis learners,
+it is equal to the opposite of the distance.
 
 >>> score = nca.pair_score([[[3.5, 3.6], [5.6, 2.4]], [[1.2, 4.2], [2.1, 6.4]], [[3.3, 7.8], [10.9, 0.1]]])
 >>> score
 array([-0.49627072, -3.65287282, -6.06079877])
 
-This is useful because `pair_score` matches the **score** semantic of 
+This is useful because ``pair_score`` matches the **score** semantic of 
 scikit-learn's `Classification metrics
 <https://scikit-learn.org/stable/modules/model_evaluation.html#classification-metrics>`_.
 
-For similarity learners `pair_distance` is not available, as they don't learn
-a distance. Intead you use `pair_score` that has the same behaviour but
-for similarity.
+For metric learners that learn a distance metric, there is also the ``pair_distance``
+method.
 
->>> algorithm.pair_score([[[3.5, 3.6], [5.6, 2.4]], [[1.2, 4.2], [2.1, 6.4]], [[3.3, 7.8], [10.9, 0.1]]])
-array([-0.2312, 705.23, -72.8])
+>>> nca.pair_distance([[[3.5, 3.6], [5.6, 2.4]], [[1.2, 4.2], [2.1, 6.4]], [[3.3, 7.8], [10.9, 0.1]]])
+array([0.49627072, 3.65287282, 6.06079877])
 
 .. warning::
     
-    If you try to use `pair_distance` with a similarity learner, an error
-    will appear, as they don't learn a distance nor a pseudo-distance.
+    If you try to use ``pair_distance`` with a bilinear similarity learner, an error
+    will be thrown, as it does not learn a distance.
+
+You can also return a function that will return the metric learned. It can
+compute the metric between two 1D arrays, similarly to distance functions in
+`scipy.spatial.distance`. To do that, use the ``get_metric`` method.
+
+>>> metric_fun = nca.get_metric()
+>>> metric_fun([3.5, 3.6], [5.6, 2.4])
+0.4962707194621285
 
-You can also call `get_metric` with similarity learners, and you will get
-a function that will return the similarity bewtween 1D arrays.
+You can also call ``get_metric`` with bilinear similarity learners, and you will get
+a function that will return the similarity between 1D arrays.
 
 >>> similarity_fun = algorithm.get_metric()
 >>> similarity_fun([3.5, 3.6], [5.6, 2.4])
 -0.04752
 
-For similarity learners and mahalanobis learners, `pair_score` is
-available. You can interpret that this function returns the **score**
-between points: the more the **score**, the closer the pairs and vice-versa.
-For mahalanobis learners, it is equal to the inverse of the distance.
+Finally, as explained in :ref:`mahalanobis_distances`, these are equivalent to the Euclidean
```

Comment:
you may need to replace "these" by "Mahalanobis distances" here depending on how the ref is displayed once the doc is compiled?

Reply:
It's rendered as "Mahalanobis distances", so it felt redundant to repeat it again. See the render [here](https://imgur.com/a/I4Iw6mb)